### PR TITLE
Added HDR10 Tone Mapping sample code including H2H and H2S config files.

### PR DESCRIPTION
**Pull request by FurongZhang**
Signed-off-by: Furong Zhang <furong.zhang@intel.com>

### CODE REVIEW

File: videoprocess/vpphdr_tm.cpp [hunk 1]
```diff
@@ -0,0 +1,861 @@
+/*
+* Copyright (c) 2009-2021, Intel Corporation
```
Comment (Tianhaol) [inline videoprocess/vpphdr_tm.cpp:2]: 2022?

Comment (FurongZhang) [reply videoprocess/vpphdr_tm.cpp]: Sure. Done.

**Comment by Tianhaol** [comment]
Same output formats in these two cfg files?  Is it enough to keep one file?

**Comment by FurongZhang** [comment]
> Same output formats in these two cfg files? Is it enough to keep one file?

Thanks @Tianhaol so much for the reviewing. The two configuration files are different.

**Comment by FurongZhang** [comment]
The difference between 2 configuration files are:
1. TM_TYPE
2. DST_FILE_NAME
3. DST_FRAME_FORMAT
4. DST_FRAME_TRANSFER_CHARACTERISTICS

**Review by Tianhaol — APPROVED** [review]

File: videoprocess/vpphdr_tm.cpp [hunk 2]
```diff
@@ -0,0 +1,861 @@
...
+static uint32_t g_in_pic_average_luminance = 1000;
+static uint32_t g_out_max_display_luminance = 1000;
+static uint32_t g_out_min_display_luminance = 1;
+static uint32_t g_out_max_content_luminance = 4000;
+static uint32_t g_out_pic_average_luminance = 1000;
+
+static uint32_t g_in_colour_primaries = 9;
+static uint32_t g_in_transfer_characteristic = 16;
+
+static uint32_t g_out_colour_primaries = 9;
```
Comment (XinfengZhang) [inline videoprocess/vpphdr_tm.cpp:89]: could we use macro or enum for these const?

File: videoprocess/vpphdr_tm.cpp [hunk 3]
```diff
@@ -0,0 +1,861 @@
...
+static uint32_t g_out_transfer_characteristic = 16;
+
+static uint32_t g_tm_type =1;
```
Comment (XinfengZhang) [inline videoprocess/vpphdr_tm.cpp:92]: same

File: videoprocess/vpphdr_tm.cpp [hunk 4]
```diff
@@ -0,0 +1,861 @@
...
+    in_hdr10_metadata.max_content_light_level         = g_in_max_content_luminance;
+    in_hdr10_metadata.max_pic_average_light_level     = g_in_pic_average_luminance;
+    in_hdr10_metadata.display_primaries_x[0] = 8500;
+    in_hdr10_metadata.display_primaries_y[0] = 39850;
+    in_hdr10_metadata.display_primaries_x[1] = 35400;
+    in_hdr10_metadata.display_primaries_y[1] = 14600;
+    in_hdr10_metadata.display_primaries_x[2] = 6550;
+    in_hdr10_metadata.display_primaries_y[2] = 2300;
+    in_hdr10_metadata.white_point_x = 15635;
+    in_hdr10_metadata.white_point_y = 16450;
```
Comment (XinfengZhang) [inline videoprocess/vpphdr_tm.cpp:214]: these values is standard meta data?

File: videoprocess/vpphdr_tm.cpp [hunk 5]
```diff
@@ -0,0 +1,861 @@
...
+    case VA_TONE_MAPPING_HDR_TO_HDR:
+        out_hdr10_metadata.display_primaries_x[0] = 8500;
+        out_hdr10_metadata.display_primaries_y[0] = 39850;
+        out_hdr10_metadata.display_primaries_x[1] = 35400;
+        out_hdr10_metadata.display_primaries_y[1] = 14600;
+        out_hdr10_metadata.display_primaries_x[2] = 6550;
+        out_hdr10_metadata.display_primaries_y[2] = 2300;
+        out_hdr10_metadata.white_point_x = 15635;
+        out_hdr10_metadata.white_point_y = 16450;
+        break;
```
Comment (XinfengZhang) [inline videoprocess/vpphdr_tm.cpp:252]: hdr to hdr case,the meta data will keep unchanged?

File: videoprocess/vpphdr_tm.cpp (hunk 2 above)
Comment (FurongZhang) [reply videoprocess/vpphdr_tm.cpp:89]: In VAAPI, https://github.com/intel/libva/blob/master/va/va_vpp.h#L715, colour_primaries is defined as uint8_t. Hence, here, I used some integer. 

Please let me know your feedback on this.

File: videoprocess/vpphdr_tm.cpp [hunk 6]
```diff
@@ -0,0 +1,861 @@
...
+static uint32_t g_out_max_display_luminance = 1000;
+static uint32_t g_out_min_display_luminance = 1;
+static uint32_t g_out_max_content_luminance = 4000;
+static uint32_t g_out_pic_average_luminance = 1000;
+
+static uint32_t g_in_colour_primaries = 9;
+static uint32_t g_in_transfer_characteristic = 16;
+
+static uint32_t g_out_colour_primaries = 9;
+static uint32_t g_out_transfer_characteristic = 16;
```
[the remark this answers is not in the document]
Comment (FurongZhang) [reply videoprocess/vpphdr_tm.cpp:90]: Same as the above.

File: videoprocess/vpphdr_tm.cpp (hunk 4 above)
Comment (FurongZhang) [reply videoprocess/vpphdr_tm.cpp:214]: Yes, these value is BT2020 standard color primaries.

File: videoprocess/vpphdr_tm.cpp (hunk 2 above)
Comment (FurongZhang) [reply videoprocess/vpphdr_tm.cpp:89]: If you think it is difficult to read, maybe need to think about a way to improve that. Please let me know your feedback. Thank you!

File: videoprocess/vpphdr_tm.cpp (hunk 5 above)
Comment (FurongZhang) [reply videoprocess/vpphdr_tm.cpp:252]: In the example, just change the luminance. Color primaries unchanged.

File: videoprocess/vpphdr_tm.cpp (hunk 2 above)
Comment (XinfengZhang) [reply videoprocess/vpphdr_tm.cpp:89]: from the your comments: https://github.com/intel/libva/commit/7bbddffddb4a15f0163f89d5d87860866e850cde
the value of primaries is :1, 6, 9 and characteristic is 1, 4, 6,8, 13, 14,15, 16.  looks hard to using a simple string to describe it

**Review by XinfengZhang — APPROVED** [review]

**Pull request merged**